### DISCUSSION TITLE
item ui offset griefing

### DISCUSSION THREAD

**Discussion by DrBruhman** [FakeFishGames/Barotrauma#14439] [Bug reports]
client 1:sets resolution to 50:1(or something realistic) and then dragging ui to screen corner
client 2:with normal resolution, observes unability to use item

could be done more easily by using lua/cs mods

**Comment by DrBruhman** [comment]
also ui refreshes only between rounds or by resolution changing/ui scaling

**Comment by Next-Door-Tech** [comment]
Assuming this does in fact work how it sounds, I could see this being done unintentionally by players using ultrawide monitors.

I haven't experienced this personally though and I can't test it myself at the moment.

**Reply by DrBruhman** [reply]
my 5:4 ultrasquire monitor is bad for competitive barogaming

**Comment by gerakis** [comment]
Does it change for other players? I was under the impression each client gets their own local copy of the coords. So client A can put it way off screen yet client B still sees it in the default place.

Was this something you have seen happen?

**Reply by DrBruhman** [reply]
yeah, i banned one guy because of this, he already seems shady and after abusing this bug i had to kill him.

**Reply by DrBruhman** [reply]
devs did it as a shortcut for not making a local buffer for items parameters, and it backfired on me

**Reply by Regalis11** [reply]
> devs did it as a shortcut for not making a local buffer for items parameters, and it backfired on me

Did we? 🤔 I thought I implemented it because people were complaining that the changes to the UI layout weren't visible to other clients.

**Reply by DrBruhman** [reply]
> > devs did it as a shortcut for not making a local buffer for items parameters, and it backfired on me
> 
> Did we? 🤔 I thought I implemented it because people were complaining that the changes to the UI layout weren't visible to other clients.

changes arent visible anyway to the point when you update ui by ending round rejoining or changing ui scale/resolution.
i remember at least 2 servers that have meet the ui offset griefers and the team was unable to do anything.

**Comment by pavel4win** [comment]
Those who play with me often face this problem. I play on three monitors (merged into one by AMD eyefinity), so the resolution is 5760x1080
If I move any window as it is convenient for me, other players stop seeing it.

To avoid this, you need to store the coordinates of the windows in virtual coordinates.

**Comment by Regalis11** [comment]
This has been fixed in our private repo now, fix coming up in a future update!
